.github/workflows/main.yml

```diff
@@ -44,7 +44,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        rust: [1.57.0, stable, beta, nightly]
+        rust: [1.63.0, stable, beta, nightly]
```

Comment:
Bolero is a bit different than cargo-fuzz/arbitrary in this area since we allow writing basic unit tests mixed in with your application code. Cargo-fuzz, OTOH, forces you to put them in completely separate crates that don't have the same compatibility requirements as your main product.

Reply:
While what you're saying is definitely right for cargo-fuzz vs. bolero, I think bolero-generator and arbitrary have basically the same goal: have all crates expose implementation of them, which means they need to be as stable as possible.

Anyway, I'll look into figuring out how to get `cargo test --features arbitrary` run with 1.63 and stay on 1.57 for the other tests when I look into it next, I'll just need to revert the changes made for clippy in addition to that :)

Reply:
> have all crates expose implementation of them, which means they need to be as stable as possible.

Yeah that's another great point to consider here!

Sorry for the extra work with the clippy stuff! But I think other than reverting it, we should be good to merge!

Reply:
Actually, it looks like miniz_oxide _just_ broke compatibility with rust versions up to 1.59.0... And we indirectly depend on that through the `backtrace` crate. I think we can just bump MSRV to 1.60.0 and call it good (the formatting feature should be available there).

Reply:
TBH I personally don't believe in MSRV :sweat_smile: If https://github.com/rust-lang/cargo/issues/9930 were implemented then I'd believe in it, but until it's done I don't think it makes sense to actually worry about it (and even once done I still don't think it'd actually be useful, but… I mean sure people want stability for the user, but the ones actually doing the program building can stomach keeping their toolchains up-to-date, and I say that as maintainer for the nixos distribution :sweat_smile: )

IMO having the MSRV being older than whatever is in debian stable doesn't make sense, and debian stable currently has 1.63 — I guess that's also the reason why arbitrary is also living with a 1.63 MSRV.

Anyway, I've reverted the clippy changes and instead fixed the CI to handle two different MSRVs depending on the feature set, so this should be ready to go! But let me know if you change your mind, I still have the tip of the branch around! :)